File: MatCode/naive_gauss.m
function x = naive_gauss(A,b)
n = length(b);
% Forward Elimination
for j = 1 : n-1
    if abs(A(j,j)) < eps; error('zero pivot encountered');
    end
    for i = j + 1 : n
        mult = A(i,j)/A(j,j);
        for k = j + 1: n
            A(i,k) = A(i,k) - mult*A(j,k);
        end
        b(i) = b(i) - mult*b(j);
    end
end

% Back Substitution
for i = n:-1:1
    for j = i+1 : n
        b(i) = b(i) - A(i,j)*x(j);
    end
    x(i) = b(i)/A(i,i);
end
x = transpose(x);
end

File: PyCode/naiveGauss.py
import numpy as np

def naiveGauss(A,b):
    n = len(b)
    x = np.zeros(n)
    for j in range (1,n-1):
        if np.abs(A[j][j]) < 0.0001:
            break
        for i in range(j+1,n):
            mult = A[i][j]/A[j][j]
            for k in range(j+1,n):
                A[i][k] = A[i][k] - mult*A[j][k]
            b[i] = b[i] - mult*b[j]
    # Back substitution
    for i in range(n,1,-1):
        for j in range(i + 1,n):
            b[i] = b[i] - A[i][j]*x[j]
        x[i] = b[i]/A[i][i]                     #Out of Range error here
    return x

